rdt/transformers/base.py

```diff
@@ -145,16 +145,17 @@ def _get_columns_data(data, columns):
 
     @staticmethod
     def _set_columns_data(data, columns_data, columns):
-        if columns_data is None:
-            return
+        if columns_data is not None:
+            if isinstance(columns_data, (pd.DataFrame, pd.Series)):
+                columns_data.index = data.index
 
-        if isinstance(columns_data, (pd.DataFrame, pd.Series)):
-            columns_data.index = data.index
+            if len(columns_data.shape) == 1:
+                data[columns[0]] = columns_data
+            else:
+                new_data = pd.DataFrame(columns_data, columns=columns)
+                data = pd.concat([data, new_data.set_index(data.index)], axis=1)
 
-        if len(columns_data.shape) == 1:
-            data[columns[0]] = columns_data
-        else:
-            data[columns] = columns_data
+        return data
```

Comment:
This change is motivated by the following [logs](https://github.com/sdv-dev/RDT/runs/7263153947?check_suite_focus=true)